lib/publiccloud/azure.pm

```diff
@@ -19,6 +19,7 @@ use testapi;
 
 has tenantid        => undef;
 has subscription    => undef;
+has username        => undef;
```

Comment:
so all 3 providers got `username` , maybe we need move it to `provider` instead declaring in each one ?

Reply:
Done.